app/src/main/java/com/edu/uni/augsburg/uniatron/MainActivity.java

```diff
@@ -44,6 +47,14 @@ protected void onCreate(Bundle savedInstanceState) {
         mTextMessage = (TextView) findViewById(R.id.message);
         BottomNavigationView navigation = (BottomNavigationView) findViewById(R.id.navigation);
         navigation.setOnNavigationItemSelectedListener(mOnNavigationItemSelectedListener);
-    }
+
+        // start the service when the app is first opened after installation
+        // rebooting will automatically start the service via BroadcastReceiverOnStateChanged class
+        // force closing via OS will automatically start the service because it's sticky
+        startService(new Intent(getBaseContext(), StepCountService.class));
```

Comment:
I think this is the reason why the service is rebooting every time the app orientation changes. Maybe there is a possibility to check, whether this is the first start of the app or not. If so, start the service once.